doc/src/vpr/placement_constraints.rst

```diff
@@ -0,0 +1,103 @@
+#. The regions on the FPGA chip to which those primitives must be constrained
+
+The file is passed as an input to VPR when running with placement constraints. When the file is read in, its information is used during the packing and placement stages of VPR. The hierarchy of the file is set up as follows.
+
+.. note:: Use the VPR :option: `--read_vpr_constraints` to specify the VPR placement constraints file that is to be loaded. 
+
+The top level tag is the ``<vpr_constraints>`` tag. This tag contains one ``<partition_list>`` tag. The ``<partition_list>`` tag can be made up of an unbounded number of ``<partition>`` tags. The ``<partition>`` tags contains all of the detailed information of the placement constraints, and is described in detail below.
+
+Partitions, Atoms, and Regions
+------------------------------
+
+Partition
+^^^^^^^^^
+
+A partition is made up of two components - a group of primitives (a.k.a. atoms) that must be constrained to the same area on the chip, and a set of one or more regions specifying where those primitives must be constrained. The information for each partition is contained within a ``<partition>`` tag, and the number of ``partition`` tags that the partition_list tag can contain is unbounded. 
+
+:req_param name:
+   A name for the partition.
+
+:req_param add_atom:
+   A tag used for adding an atom primitive to the partition.
+
+:req_param add_region:
+   A tag used for specifying a region for the partition.
+
+Atom 
+^^^^
+
+An ``<add_atom>`` tag is used to add an atom that must be constrained to the partition. Each partition can contain any number of atoms from the circuit. The ``<add_atom>`` tag has the following attribute:
+
+:req_param name_pattern:
+   The name of the atom.
+
+The ``name_pattern`` can either be the exact name of the atom from the atom netlist produced by VPR. It can also be a regular expression, in which case VPR will add all atoms from the netlist which have a portion of their name matching the regular expression to the partition.
+
+Region
+^^^^^^
+
+An ``<add_region>`` tag is used to add a region to the partition. A ``region`` is a rectangular area on the chip. A partition can contain any number of independent regions - the regions within one partition must not overlap with each other (in order to ease processing when loading in the file). An ``<add_region>`` tag has the following attributes.
+
+:req_param x_low:
+   The x value of the lower left point of the rectangle.
+
+:req_param y_low:
+   The y value of the lower left point of the rectangle.
+
+:req_param x_high:
+   The x value of the upper right point of the rectangle.
+
+:req_param y_high:
+   The y value of the upper right point of the rectangle.
+
+:opt_param subtile:
+   Each x, y location on the grid may contain multiple locations known as subtiles. This paramter is an optional value specifying the subtile location that the atom(s) of the partition shall be constrained to.
```

Comment:
This option is commonly used when constraining an atom to specific location (e.g. an exact I/O location); it is legal to use with larger regions, but uncommon.